email-template/src/main/java/io/micronaut/email/template/EmailTemplateCourier.java

```diff
@@ -0,0 +1,78 @@
+/*
+ * Copyright 2017-2021 original authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.micronaut.email.template;
+
+import io.micronaut.core.annotation.NonNull;
+import io.micronaut.core.annotation.Nullable;
+import io.micronaut.email.Recipient;
+import io.micronaut.email.Sender;
+import io.micronaut.views.ModelAndView;
+
+import javax.validation.Valid;
+import javax.validation.constraints.NotBlank;
+import javax.validation.constraints.NotNull;
+
+/**
+ * Contract for sending an Email template.
+ *
+ * @author Sergio del Amo
+ * @since 1.0
+ * @param <T> The model type
+ */
+@FunctionalInterface
+public interface EmailTemplateCourier<T> {
```

Comment:
I would decouple the sending and rendering part completely as there might be multiple implementation of the renderers and multiple implementation of email senders. there could be just a simple interface 

```
interface EmailRenderer {
    Email render(Email templatedEmail, Object model)
}
```

which can be later used as

```
sender.send(renderer.render(email))
```